### ISSUE TITLE
Usability issue with invoice confirmation checkbox on checkout

### ISSUE THREAD

**Issue by ben72** [Billmate/woocommerce#313]
We find that quite some customers don't check the checkbox to allow invoice to their email and get confused when they get the error message "Vänligen bekräfta att e-postadressen "xxx@yyy.se" är korrekt. Denna kommer att användas för fakturering."

The result is frustrating customers and lost sales!

Ex. Today I got a phone call where the customer said they tried to write their email address several times but it just says it's not valid every time!
And we get several Cancelled orders per month which indicates this could be a problem. An order is created when this happens but it times out and gets the state Cancelled if the customer does not check the box and continue. I tested this myself today.

I'm not sure what it should say but I'm sure your usability experts will think of something better than the current text!
My suggestion is:
"Vänligen bekräfta att e-postadressen "xxx@yyy.se" är korrekt genom att kryssa i rutan under valt betalsätt nedan."

**Comment by DanielMarklund** [comment]
> We find that quite some customers don't check the checkbox to allow invoice to their email and get confused when they get the error message "Vänligen bekräfta att e-postadressen "xxx@yyy.se" är korrekt. Denna kommer att användas för fakturering."

Ah, so this is why we keep getting those emails! I've always wondered why as I've never stumbled upon the issue myself. We've had to make manual orders via Swish when these issues have been appearing, or straight up gotten a follow up e-mail that they've done the purchase somewhere else.

**Comment by ben72** [comment]
@DanielMarklund Yeah, I never knew either until I tested myself. :)
Test it yourself and check the error message. Not very informative imho. Of course the customer thinks the email address is incorrect when they see that message and don't think about checking the checkbox to allow invoices to their email address.

**Comment by stefanliden** [comment]
Thanks @DanielMarklund  @ben72 

We are in the process of fixing the confusion for the customers. I will come back when I have more information.

**Comment by stefanliden** [comment]
Hi,  In our latest version of Billmate 3.9.1, we have now made improvements to this to create less confusion.

**Comment by ben72** [comment]
Thanks @stefanliden I will test it soon!

**Comment by ben72** [comment]
@stefanliden I have tested your changes and I'm not impressed!
It's great that you tried to make a change but why not make it the same way as WooCommerce already does for the terms & conditions checkbox?
When you don't click that checkbox, the page scrolls up for an informative message and the checkbox has a red border around it to hi-light the issue.
With your fix you interfere with the terms & conditions. That text is now put under the Billmate payment option and if you click the button to confirm the order multiple times the text is output multiple times. It looks very bad. Please see screenshots.

The text is output multiple times here when the users clicks the confirm order button. Not pretty!
![Screenshot_2020-02-08 Kassan](https://user-images.githubusercontent.com/1370224/74098434-60969100-4b18-11ea-982d-d3c6be7f5134.png)

You interfere with the terms&conditions error text here.
![Screenshot_2020-02-08 Kassan2](https://user-images.githubusercontent.com/1370224/74098435-62f8eb00-4b18-11ea-8e66-b5b1cea84e3d.png)

Please use the standard way to notify about issues in the checkout.

**Comment by ben72** [comment]
Can you please improve the usability here? Look at how it's done for the terms checkbox.
See screenshot of how it could look for Billmate as well if done like that.

![Screenshot_2020-07-07 Kassan](https://user-images.githubusercontent.com/1370224/86817839-e269f680-c085-11ea-926d-416b152a9525.png)

**Comment by MattiasNilsson** [comment]
@ben72 Thanks! We will implement this change.

**Comment by ben72** [comment]
@MattiasNilsson Thank you for your efforts! But you did not add the red border around the billmate checkbox. That alone would be a great usability improvement! It should be done similar to the checkbox for the shop terms.

Scrolling to the checkbox can possibly be OK but when you override things like this the error message for not accepting the shop terms is also displayed below the Billmate section which is not intuitive.
If you would instead replace the error message with something understandable and mark the checkbox with the red border you could skip the scrolling and just follow the WooCommerce standard way of doing things.
Also, now after clicking three times on the checkout button you get the message displaying three times which does not look pretty, see screenshot.
In short, I think you should follow WooCommerce ui standards and not make up your own.

![billmate-issue](https://user-images.githubusercontent.com/1370224/91530576-a4fa4a80-e90b-11ea-8a9b-7a8e808e31b8.png)
